modules/service/src/main/scala/gem/Service.scala

```diff
@@ -27,6 +27,14 @@ final class Service[M[_]: Sync: LiftIO] private (private val xa: Transactor[M],
       ProgramDao.selectBySubstring(s"%$substr%", max).transact(xa)
     }
 
+  /**
+   * Constuct a program that yields the `Observation` with the matching observation id.
+   */
+  def queryObservationById(id: Observation.Id): M[Observation] =
```

Comment:
I think we discussed in the chatter a bit whether assuming that the matching observation exists is appropriate but I couldn't find it again.  This should probably explicitly return an `Option[Observation]` instead?